src/ProxyManager/ProxyGenerator/Util/PublicScopeSimulator.php

```diff
@@ -147,30 +149,32 @@ private static function getTargetObject(PropertyGenerator $valueHolder = null) :
         return 'unserialize(sprintf(\'O:%d:"%s":0:{}\', strlen(get_parent_class($this)), get_parent_class($this)))';
     }
 
-    /**
-     * @param string      $operationType
-     * @param string      $nameParameter
-     * @param string|null $valueParameter
-     *
-     * @return string
-     *
-     * @throws \InvalidArgumentException
-     */
-    private static function getOperation(string $operationType, string $nameParameter, $valueParameter) : string
-    {
+    private static function getOperation(
+        ?\ReflectionMethod $originalMethod,
+        string $operationType,
+        string $nameParameter,
+        ?string $valueParameter
+    ) : string {
         switch ($operationType) {
             case static::OPERATION_GET:
-                return 'return $targetObject->$' . $nameParameter . ';';
+                return ProxiedMethodReturnExpression::generate('$targetObject->$' . $nameParameter, $originalMethod);
             case static::OPERATION_SET:
                 if (! $valueParameter) {
                     throw new \InvalidArgumentException('Parameter $valueParameter not provided');
                 }
 
-                return 'return $targetObject->$' . $nameParameter . ' = $' . $valueParameter . ';';
+                return ProxiedMethodReturnExpression::generate(
+                    '$targetObject->$' . $nameParameter . ' = $' . $valueParameter,
+                    $originalMethod
+                );
             case static::OPERATION_ISSET:
-                return 'return isset($targetObject->$' . $nameParameter . ');';
+                return ProxiedMethodReturnExpression::generate(
+                    'isset($targetObject->$' . $nameParameter . ')',
+                    $originalMethod
+                );
             case static::OPERATION_UNSET:
-                return 'unset($targetObject->$' . $nameParameter . ');';
+                return 'unset($targetObject->$' . $nameParameter . ');'
```

Comment:
`return unset($foo)` is not a valid expression, so I simply add a fake return block afterwards.